### PR TITLE
Fix viewIndex

### PR DESCRIPTION
**Pull request by AMD-dwang**
When we use gl_Layer/gl_ViewIndex in Fs, we must export them. currently,
we just handle them in vertex process stage or enabling mulit-view.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit e69756823cdd6d45d4e2b36982ef3c32b02e525f
<details><summary>CTS tests (Failed: 0/187060)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36070/64528 (55.9%)</li>
<li>Failed: 0/64528 (0.0%)</li>
<li>Not Supported: 28458/64528 (44.1%)</li><li>Warnings: 0/64528 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31532/58004 (54.4%)</li>
<li>Failed: 0/58004 (0.0%)</li>
<li>Not Supported: 26472/58004 (45.6%)</li><li>Warnings: 0/58004 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37520/64528 (58.1%)</li>
<li>Failed: 0/64528 (0.0%)</li>
<li>Not Supported: 27008/64528 (41.9%)</li><li>Warnings: 0/64528 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3c192e636f91960530674f97baf5131c18a7a8f7
<details><summary>CTS tests (Failed: 0/187060)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36070/64528 (55.9%)</li>
<li>Failed: 0/64528 (0.0%)</li>
<li>Not Supported: 28458/64528 (44.1%)</li><li>Warnings: 0/64528 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31532/58004 (54.4%)</li>
<li>Failed: 0/58004 (0.0%)</li>
<li>Not Supported: 26472/58004 (45.6%)</li><li>Warnings: 0/58004 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37520/64528 (58.1%)</li>
<li>Failed: 0/64528 (0.0%)</li>
<li>Not Supported: 27008/64528 (41.9%)</li><li>Warnings: 0/64528 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchInOutImportExport.cpp
Comment (kuhar) [inline lgc/patch/PatchInOutImportExport.cpp:1230]: nit: llvm prefers `!m_layer`, i.e.: `if (useLayer && !m_layer)`

File: lgc/patch/PatchInOutImportExport.cpp
Comment (kuhar) [inline lgc/patch/PatchInOutImportExport.cpp:1229]: I think you can do `useLayer |= nextBuiltInUsage.layer || nextBuiltInUsage.viewIndex`

Comment (kuhar) [reply lgc/patch/PatchInOutImportExport.cpp:1229]: Also a few lines above in the existing code

File: lgc/patch/PatchInOutImportExport.cpp
Comment (kuhar) [inline lgc/patch/PatchInOutImportExport.cpp:1479]: The type is not obvious to me here, could you use the exact type instead of `auto`?

File: lgc/patch/PatchInOutImportExport.cpp
Comment (kuhar) [inline lgc/patch/PatchInOutImportExport.cpp:1483]: You can do `assert(is_contained(set, element))`. Also everywhere else.

File: lgc/patch/PatchInOutImportExport.cpp
Comment (kuhar) [inline lgc/patch/PatchInOutImportExport.cpp:1497]: Was this line removed on accident?

**Review by kuhar — COMMENTED** [review]
Just some nits.

File: lgc/patch/PatchInOutImportExport.cpp
Comment (AMD-dwang) [reply lgc/patch/PatchInOutImportExport.cpp:1483]: Currently this format is heavily used, I think another change might be better. Thanks.

File: lgc/patch/PatchInOutImportExport.cpp
Comment (AMD-dwang) [reply lgc/patch/PatchInOutImportExport.cpp:1479]: This is an unnamed structure.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 85c0058cfaa287cad53397195cbb06b4adf9dd0d
<details><summary>CTS tests (Failed: 2/173900)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36645/65183 (56.2%)</li>
<li>Failed: 0/65183 (0.0%)</li>
<li>Not Supported: 28538/65183 (43.8%)</li><li>Warnings: 0/65183 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26304/43534 (60.4%)</li>
<li><details><summary>Failed: 1/43534 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.subgroups.ballot.ext_shader_subgroup_ballot.compute.compute
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 17229/43534 (39.6%)</li><li>Warnings: 0/43534 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37802/65183 (58.0%)</li>
<li><details><summary>Failed: 1/65183 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.synchronization.basic.event.multi_secondary_command_buffer
Stack trace: Script:
synchronizationWrapper->queueSubmit(queue, *fence): VK_TIMEOUT at vktSynchronizationBasicEventTests.cpp:337

```
</p></details></li>

<li>Not Supported: 27380/65183 (42.0%)</li><li>Warnings: 0/65183 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 5b6625654e6e3bfd3d0ec1ef4ea02863ddc982b7
<details><summary>CTS tests (Failed: 1/187694)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36645/65183 (56.2%)</li>
<li>Failed: 0/65183 (0.0%)</li>
<li>Not Supported: 28538/65183 (43.8%)</li><li>Warnings: 0/65183 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31111/57328 (54.3%)</li>
<li>Failed: 0/57328 (0.0%)</li>
<li>Not Supported: 26217/57328 (45.7%)</li><li>Warnings: 0/57328 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37802/65183 (58.0%)</li>
<li><details><summary>Failed: 1/65183 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.synchronization.basic.event.multi_secondary_command_buffer
Stack trace: Script:
synchronizationWrapper->queueSubmit(queue, *fence): VK_TIMEOUT at vktSynchronizationBasicEventTests.cpp:337

```
</p></details></li>

<li>Not Supported: 27380/65183 (42.0%)</li><li>Warnings: 0/65183 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 5b6625654e6e3bfd3d0ec1ef4ea02863ddc982b7
<details><summary>CTS tests (Failed: 1/187820)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36645/65225 (56.2%)</li>
<li>Failed: 0/65225 (0.0%)</li>
<li>Not Supported: 28580/65225 (43.8%)</li><li>Warnings: 0/65225 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31111/57370 (54.2%)</li>
<li>Failed: 0/57370 (0.0%)</li>
<li>Not Supported: 26259/57370 (45.8%)</li><li>Warnings: 0/57370 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37802/65225 (58.0%)</li>
<li><details><summary>Failed: 1/65225 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.synchronization.basic.event.multi_secondary_command_buffer
Stack trace: Script:
synchronizationWrapper->queueSubmit(queue, *fence): VK_TIMEOUT at vktSynchronizationBasicEventTests.cpp:337

```
</p></details></li>

<li>Not Supported: 27422/65225 (42.0%)</li><li>Warnings: 0/65225 (0.0%)</li>
</ul></details>


</ul></details>

**Review by kuhar — DISMISSED** [review]
The code looks fine to me. I didn't look for semantic issues, so please get a second review from someone more familiar with this part of LGC.

File: lgc/patch/PatchInOutImportExport.cpp
Comment (amdrexu) [inline lgc/patch/PatchInOutImportExport.cpp:1232]: PS -> FS

**Review by amdrexu — DISMISSED** [review]
LGTM

**Review by amdrexu — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit e88cfe29a69b8d81298d2f959ebddf6140bd8163
<details><summary>CTS tests (Failed: 1/187820)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36645/65225 (56.2%)</li>
<li>Failed: 0/65225 (0.0%)</li>
<li>Not Supported: 28580/65225 (43.8%)</li><li>Warnings: 0/65225 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31111/57370 (54.2%)</li>
<li>Failed: 0/57370 (0.0%)</li>
<li>Not Supported: 26259/57370 (45.8%)</li><li>Warnings: 0/57370 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37802/65225 (58.0%)</li>
<li><details><summary>Failed: 1/65225 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.synchronization.basic.event.multi_secondary_command_buffer
Stack trace: Script:
synchronizationWrapper->queueSubmit(queue, *fence): VK_TIMEOUT at vktSynchronizationBasicEventTests.cpp:337

```
</p></details></li>

<li>Not Supported: 27422/65225 (42.0%)</li><li>Warnings: 0/65225 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit ef3b4a33dd905b07de19254b04ced41c96c0f46e
<details><summary>CTS tests (Failed: 1/187820)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36645/65225 (56.2%)</li>
<li>Failed: 0/65225 (0.0%)</li>
<li>Not Supported: 28580/65225 (43.8%)</li><li>Warnings: 0/65225 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31111/57370 (54.2%)</li>
<li>Failed: 0/57370 (0.0%)</li>
<li>Not Supported: 26259/57370 (45.8%)</li><li>Warnings: 0/57370 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37802/65225 (58.0%)</li>
<li><details><summary>Failed: 1/65225 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.synchronization.basic.event.multi_secondary_command_buffer
Stack trace: Script:
synchronizationWrapper->queueSubmit(queue, *fence): VK_TIMEOUT at vktSynchronizationBasicEventTests.cpp:337

```
</p></details></li>

<li>Not Supported: 27422/65225 (42.0%)</li><li>Warnings: 0/65225 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Pull request merged**
